nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/serialization/VersionedFlowSynchronizer.java

```diff
@@ -378,6 +378,7 @@ private void synchronizeFlow(final FlowController controller, final DataFlow exi
 
             if (versionedFlow != null) {
                 controller.setMaxTimerDrivenThreadCount(versionedFlow.getMaxTimerDrivenThreadCount());
+                controller.setMaxEventDrivenThreadCount(versionedFlow.getMaxTimerDrivenThreadCount());
```

Comment:
This should be `.getMaxEventDrivenThreadCount()` not `.getMaxTimerDrivenThreadCount()`

Reply:
The `getMaxEventDrivenThreadCount()` also will default to `0`, though. So if the flow.json.gz doesn't have that field (which will be everyone's when they are upgrading) it will call `controller.setMaxEventDrivenThreadCount(0)` which will throw an `IllegalArgumentException`. So need to either default the value to `1` or check if it's set before calling this.

Reply:
I updated and set the maxEventDrivenThreadCount = 1 in the VersionedDataflow. Wasn't sure if this was hacky or if there's a better place to set that default.